Support for multiple "from"-fields with the same value.

bioxz opened the pull request:
The release of Factorio 0.14.0 added two update possibilities for 0.13.18: 0.13.18 to 0.13.19 and 0.13.18 to 0.14.0. This commit supports that change by respecting the experimental flag earlier. Due to those changes, some declarations are moved to the top of the function and some comments have been added.

narc0tiq commented on update_factorio.py:
```diff
@@ -51,25 +51,30 @@ def get_updater_data(user, token):
-    updates = []
+    # Get available updates
+    for row in updater_json[factorio_package]:
+        if not experimental and 'from' in row and row['from'] >= current_version and row['to'] <= latest[0]:
```
Ok, it's been a fair while since I wrote this originally, but I'm pretty sure comparing versions as strings doesn't work correctly ("0.13.2" compares as > "0.13.10"); this is why the `version_key` function (stolen from elsewhere, IIRC) exists and breaks the string into a list of ints (which does compare as expected).

Otherwise, I expect this does as expected, so fix the version comparison and I'll go ahead and merge it.

narc0tiq commented:
Thanks for the assist, I really appreciate it.

bioxz commented:
Seems like i also removed the return value latest[1] by accident, which is used later to show the latest experimental version. I've added some lines to get that value.

narc0tiq commented:
Looks good to me, let me get this merged.
